guides/source/action_mailbox_basics.md

```diff
@@ -277,13 +294,65 @@ class ApplicationMailbox < ActionMailbox::Base
+
+```bash
+$ mail
+#<Mail::Message:33780, Multipart: false, Headers: <Date: Wed, 31 Jan 2024 22:18:40 -0600>, <From: someone@hey.com>, <To: save@example.com>, <Message-ID: <65bb1ba066830_50303a70397e@Bhumis-MacBook-Pro.local.mail>>, <In-Reply-To: >, <Subject: Hello Action Mailbox>, <Mime-Version: 1.0>, <Content-Type: text/plain; charset=UTF-8>, <Content-Transfer-Encoding: 7bit>, <x-original-to: >>
+$ mail.to
+["save@example.com"]
+$ mail.from
+["someone@hey.com"]
+$ mail.date
+Wed, 31 Jan 2024 22:18:40 -0600
+$ mail.subject
+"Hello Action Mailbox"
+$ mail.body
+#<Mail::Body:0x00007fc74cbf46c0 @boundary=nil, @preamble=nil, @epilogue=nil, @charset="US-ASCII", @part_sort_order=["text/plain", "text/enriched", "text/html", "multipart/alternative"], @parts=[], @raw_source="This is the body of the email message.", @ascii_only=true, @encoding="7bit">
+$ mail.body.decoded
+"This is the body of the email message."
+# mail.decoded, a shorthand for mail.body.decoded, also works
+$ mail.decoded
+"This is the body of the email message."
+```
+
+### Inbound Email Status
+
+Once an email has been routed to a matching mailbox and processed, Action Mailbox updates the email status stored in `action_mailbox_inbound_emails` table with one of the following values:
+
+- `pending`: Received by one of the ingress controllers and scheduled for routing.
+- `processing`: During active processing, while a specific mailbox is running its `process` method.
+- `delivered`: Successfully processed by the specific mailbox.
+- `failed`: An exception was raised during the specific mailbox’s execution of the `process` method.
+- `bounced`: Rejected processing by the specific mailbox and bounced to sender.
+
+If the email is marked either `delivered`, `failed`, or `bounced` it's considered "processed" and marked for [incineration](#incineration-of-inboundemails).
+
+## Example
+
+Here is an example of an Action Mailbox that processes emails to create "forwards" for the user's project.
+
+The `before_processing` callback is used to ensure that certain conditions are met before `process` method is called. In this case, `before_processing` checks that the user has at least one project. Other supported [Action Mailbox callbacks](https://api.rubyonrails.org/classes/ActionMailbox/Callbacks.html) are `after_processing` and `around_processing`.
+
+The email can be bounced using `bounced_with` if the "forwarder" has no projects. The "forwarder" is a `User` with the same email as `mail.from`.
+
+If the "forwarder" does have at least one project, the `record_forward` method creates an Active Record model in the application using the email data `mail.subject` and `mail.body.decoded`. Otherwise, it sends an email, using Action Mailer, requesting the "forwarder" to choose a project.
```

Comment:
Minor: maybe we just use the shortcut version, `mail.decoded`?
```suggestion
If the "forwarder" does have at least one project, the `record_forward` method creates an Active Record model in the application using the email data `mail.subject` and `mail.decoded`. Otherwise, it sends an email, using Action Mailer, requesting the "forwarder" to choose a project.
```